### PR TITLE
optimize: use sonic instead of encoding/json

### PR DESCRIPTION
**Pull request by jkskj**
#### What type of PR is this?

optimize

#### What this PR does / why we need it (English/Chinese):

use sonic instead of encoding/json

### CODE REVIEW

**Comment by jkskj** [comment]
I have no idea about why some checks were not successful.

**Comment by GuangmingLuo** [comment]
have you try `gofumpt -l -extra .` ?

**Comment by GuangmingLuo** [comment]
update .github/workflows/pr-check.yml 
apache/skywalking-eyes@main -> apache/skywalking-eyes@v4

**Review by GuangmingLuo — APPROVED** [review]

File: json_test.go [hunk 1]
```diff
@@ -82,7 +85,7 @@ func TestPartialJSONRead(t *testing.T) {
 
 	for i := 0; i < messageCount; i++ {
 		err := rc.ReadJSON(&v)
-		if err != io.ErrUnexpectedEOF {
+		if err != io.ErrUnexpectedEOF && !errors.As(err, &decoder.SyntaxError{}) {
```
Comment (Duslia) [inline json_test.go:88]: Why add this condition?

Comment (jkskj) [reply json_test.go:88]: In the test cases,`encoding/json` return `io.EOF` while `sonic` return `SyntaxError`.

**Review by Duslia — APPROVED** [review]

**Pull request merged**